### DISCUSSION TITLE
Doubts on Invoke-WindowsNativeAppWithElevation function.

### DISCUSSION THREAD

**Discussion by kvprasoon** [PowerShell/Crescendo#83] [Q&A] [answered]
Reposting as no response on the blog reply :-) .

I think Crescendo should be providing option to have -Credential to the cmdlets it creates using the same Parameter attributes via json,
which can utilize the Invoke-WindowsNativeAppWithElevation. Otherwise its either using a secret store or purely interactive(correct me if I’m wrong).

Is there any reason why the elevation function looks like a cmdlet (Invoke-WindowsNativeAppWithElevation) than a simple usage like sudo for linux. If that cmdlet is always used internally, then can’t we make it simple in name ?

**Comment by theJasonHelmick** [comment, marked as answer]
Hey @kvprasoon , thanks for the suggestion.  Invoke-WindowsNativeAppWithElevation is a Crescendo function that calls Start-Process with -Credential. Why is named like a cmdlet? Cause it is ;)   We plan on sticking with this through GA. Keep in mind, that the elevation command for Windows `Invoke-WindowsNativeAppWithElevation` can be replaced with your own. We wanted you to have the flexibility to replace this with your own elevation mechanism.
